### PR TITLE
Docker image

### PR DESCRIPTION
**Pull request by hermanjunge**
* Image size is `539.8 MB`.
* This aims to be the production repository image: It will get the `raiden` contents from `HEAD`.
* User should adjust rev required in `Dockerfile` if a specific one is needed.

## Building the image

    docker build -t raiden-network/raiden -f docker/Dockerfile .

## Running the container

    docker run -ti --rm raiden-network/raiden <parameters>

### CODE REVIEW

**Comment by utzig** [comment]
I've been doing tests with this image and it LGTM.

**Comment by hermanjunge** [comment]
There a couple of things you need to take into consideration before merging this PR.

* Documentation on how to run the image.i.e. `--net=host`, `-p 80:8080`, volumes? parameters, etc.
* You need to provide the public with the ready image in `hub.docker.com`
* Now, you can promote that raiden can run in windows! :tada:

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    && apt-get dist-upgrade -q -y \
+    && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 923F6CA9 \
+    && echo "deb http://ppa.launchpad.net/ethereum/ethereum/ubuntu xenial main" \
+       | tee -a /etc/apt/sources.list.d/ethereum.list \
+    && apt-get update -q -y \
+    && apt-get install -y automake build-essential git-core libffi-dev \
+       libgmp-dev libssl-dev libtool pkg-config python-dev python-pip solc \
+    && pip install --upgrade pip \
+    && rm -rf /var/lib/apt/lists/* \
+    && git clone https://github.com/ethereum/pyrlp /apps/pyrlp \
```
Comment (utzig) [inline docker/Dockerfile]: Adding `--depth=1` to all those clones would make the build a lot faster (also we don't need the history).

**Comment by konradkonrad** [comment]
Thanks a lot @hermanjunge, that is a good start!

Two notes on what I would want to have changed:
1) I prefer to use stable/released dependencies, so all the `git clone`s for `https://github.com/ethereum/...`  projects can be removed.
2) Also I happily trade space for some build caching plus readability, so the `apt` system dependency installations (`solc`, `python`, build-tools) can become a first `RUN` block, while cloning and `pip -e install .`ing the raiden repository can become a second step. In a preliminary experiment with that, the image size ended up being ~670 MB.

> Documentation on how to run the image. i.e. --net=host, -p 80:8080, volumes? parameters, etc.

I think we can default `EXPOSE` port `40001`.
Ethereum-rpc provider can either be `--link`ed into the container network (e.g. when using `ethereum/client-go`), or, if they are off-site, be contacted by hostname. And yeah, if the ethereum node runs native on the host, then `--net=host` is the way to go.

**Comment by utzig** [comment]
I will merge this and open a new PR with cleanups and suggested updates

**Pull request merged**